src/native/minipal/cpufeatures.c

```diff
@@ -277,6 +277,28 @@ int minipal_getcpufeatures(void)
                                             {
                                                 result |= XArchIntrinsicConstants_AvxVnni;
                                             }
+
+                                            if ((cpuidInfo[CPUID_EDX] & (1 << 19)) != 0)                                // Avx10
+                                            {
+                                                __cpuidex(cpuidInfo, 0x00000024, 0x00000000);
+                                                const int versionMask = 0xFF; // [7:0]
+                                                if((cpuidInfo[CPUID_EBX] & versionMask) >= 1) // version higher than 1
+                                                {
+                                                    result |= XArchIntrinsicConstants_Avx10v1;
+                                                }
+
+                                                const int vector256Mask = (1 << 17);
+                                                const int vector512Mask = (1 << 18);
+                                                if((cpuidInfo[CPUID_EBX] & vector256Mask) != 0)
+                                                {
+                                                    result |= XArchIntrinsicConstants_Avx10v1_V256;
+                                                }
+
+                                                if((cpuidInfo[CPUID_EBX] & vector512Mask) != 0)
+                                                {
+                                                    result |= XArchIntrinsicConstants_Avx10v1_V512;
+                                                }
```

Comment:
nit: These can only be considered if `AVX10/V128` is set and if the `version` was greater than `1`, so we should have them appropriately nested